tests/test_chat.py

```diff
@@ -46,35 +49,20 @@ def multinomial(*_, **__):
     actual = chat.generate(model, input_idx, max_returned_tokens, stop_tokens=stop_tokens)
     actual = list(actual)
 
-    assert len(actual) == len(expected)
+    assert len(actual) == len(expected), (actual, expected)
     if not actual:
-        assert actual == expected
+        assert actual == expected, (actual, expected)
     else:
         for t in actual:
-            assert t.dtype == torch.long
-        assert torch.cat(actual).tolist() == expected
+            assert t.dtype == torch.long, t.dtype
+        actual_list = torch.cat(actual).tolist()
+        assert actual_list == expected, (actual_list, expected)
 
 
 @pytest.mark.parametrize("tokenizer_backend", ["huggingface", "sentencepiece"])
 def test_decode(tokenizer_backend):
-    class Tokenizer:
-        backend = tokenizer_backend
-        id2token = {1: "foo ", 2: "bar ", 3: "baz "}
-
-        def decode(self, tensor: torch.Tensor) -> str:
-            tensor = [tensor] if tensor.ndim == 0 else tensor
-            return "".join(self.id2token[int(value)] for value in tensor)
-
-    tokenizer_mock = Tokenizer()
-
-    fabric = Fabric(devices=1, accelerator="cpu")
-
-    token_stream = torch.tensor([3, 2, 1])
-    out, err = StringIO(), StringIO()
-    with redirect_stdout(out), redirect_stderr(err):
-        chat.decode(fabric, tokenizer_mock, token_stream)
-
-    assert out.getvalue() == "baz bar foo "
+    # TODO: Rewrite this test. Load a tokenizer from a checkpoint dir, and test decode() and decode_stream().
```

Comment:
Just marking it so it doesn't slip through the cracks

Reply:
Rewrote the test.